internal/cas/git.go

```diff
@@ -0,0 +1,237 @@
+package cas
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	minGitPartsLength = 2
+)
+
+// GitRunner handles git command execution
+type GitRunner struct {
+	WorkDir string
+}
+
+// NewGitRunner creates a new GitRunner instance
+func NewGitRunner() *GitRunner {
+	return &GitRunner{}
+}
+
+// WithWorkDir returns a new GitRunner with the specified working directory
+func (g *GitRunner) WithWorkDir(workDir string) *GitRunner {
+	// Create new instance instead of modifying existing one
+	return &GitRunner{
+		WorkDir: workDir,
+	}
+}
+
+// RequiresWorkDir returns an error if no working directory is set
+func (g *GitRunner) RequiresWorkDir() error {
+	if g.WorkDir == "" {
+		return &WrappedError{
+			Op:      "git",
+			Context: "no working directory set",
+			Err:     ErrNoWorkDir,
+		}
```

Comment:
```suggestion
                 err := &WrappedError{
			Op:      "git",
			Context: "no working directory set",
			Err:     ErrNoWorkDir,
		}
		return errors.New(err)
```

The suggested applies to all similar cases.